books/doc/practices.lisp

```diff
@@ -27,28 +27,154 @@
+               blank line.) This is a posix-standard. Most editors should
+               handle this automatically."))
+   (xdoc::h3 "Comments")
+   (xdoc::ul
+    (xdoc::li "Comments should begin with a single space. E.g.:"
+              (xdoc::codeblock "; This is a comment.")
+              (xdoc::i "not")
+              (xdoc::codeblock ";This is a comment."))
+    (xdoc::li "Books should begin with a copyright and licensing header (which
+               may just point to the repository's top-level LICENSE file).")
+    (xdoc::li "Inline comments which are intended to be indented at the same
+               level as a form would be at a given position should begin with
+               two semicolons. This is a standard practice in Common Lisp, and
+               editors like Emacs will indent such comments appropriately.")
+    (xdoc::li "Avoid commented code which would disrupt the s-expression
+               balance if uncommented. For instance, instead of this:"
+              (xdoc::codeblock
+               "(let ((x foo)"
+               "      ;; (y bar))"
+               "      )"
+               "  baz"
+               "  )")
+              "do this:"
+              (xdoc::codeblock
+               "(let ((x foo)"
+               "      ;; (y bar)"
+               "      )"
+               "  baz"
+               "  )"))
+    (xdoc::li "Consider whether comments would be more appropriate as "
+              (xdoc::seetopic "xdoc" "XDOC")
+              " documentation."))
+   (xdoc::h3 "Documentation")
+   (xdoc::ul
+    (xdoc::li (xdoc::seetopic "xdoc" "XDOC")
+              " short forms should end with a period, even if they are not a
+               full sentence."))
+   (xdoc::h3 "Naming Conventions")
+   (xdoc::ul
+    (xdoc::li "Consider ending predicates with ``p'' (or ``-p'', especially if
+               needed for clarity). This is a convention inherited from Common
+               Lisp.")
+    (xdoc::li "For ordinary identifiers, avoid uppercase and use hyphens to
+               delimit words (a style sometimes called ``dash-case'',
+               ``kebab-case'', or ``lisp-case''). E.g., ``my cool function''
+               would be rendered @('my-cool-function').")
+    (xdoc::li "See @(see naming-rewrite-rules) for recommendations regarding
+               the naming of rewrite rules."))
+   (xdoc::h3 "Formatting")
+   (xdoc::ul
+    (xdoc::li "Multiline @('if') forms should be formatted as follows:"
+              (xdoc::codeblock
+               "(if x"
+               "    y"
+               "  z)")
+              "That is, 4 spaces for the ``then'' term, and 2 spaces for the
+               ``else'' term. Emacs may indent this incorrectly by default."))
+   (xdoc::h3 "Packages")
+   (xdoc::ul
+    (xdoc::li "The use of packages is highly encouraged if you expect your
+               library to be used by others. This helps to avoid name clashes
```

Comment:
... or if you include your library in the manual.

Reply:
Up to you. I guess it's good to give motivation, but on the other hand other best practices are not explicitly motivated.

Reply:
I would still keep the "This helps to avoid name clashes" part, but drop the "if you expect your library to be used by others".

Reply:
Yes, that covers it all.